open-hackathon-adminUI/src/app/admin/admin_mgr.py

```diff
@@ -138,21 +119,41 @@ def get_admin_info(self, admin):
             "last_login_time": str(admin.last_login_time)
         }
 
-#    def get_admin_detail_info(self, admin, **kwargs):
-#        detail = self.get_admin_info(admin)
-#
-#        experiments = admin.experiments.filter_by(status=ExprStatus.Running).all()
-#        detail["experiments"] = []
-#        map(lambda e: detail["experiments"].append({
-#            "id": e.id,
-#            "hackathon_id": e.hackathon_id
-#        }), experiments)
-#        detail["register_state"] = True
-#        hack = hack_manager.get_hackathon_by_name(kwargs['hackathon_name'])
-#        if hack is not None and safe_get_config('checkRegister', False) == True:
-#            detail["register_state"] = self.__validate_admin_registered(admin, hack)
-
-#        return detail
+    def get_hackid_from_adminid(self, admin_id):
+
+        # can not use backref in db models
+
+        # get emails from admin though admin.id in table admin_email
+        admin_emails = self.db.find_all_objects_by(AdminEmail, admin_id=admin_id)
+        emails = map(lambda x: x.email, admin_emails)
+
+        # get AdminUserHackathonRels from query withn filter by email
+        admin_user_hackathon_rels = self.db.find_all_objects(AdminUserHackathonRel,
+                                                             AdminUserHackathonRel.admin_email.in_(emails))
+        if admin_user_hackathon_rels is None:
+            return None
+
+        # get hackathon_ids_from AdminUserHackathonRels details
+        hackathon_ids = map(lambda x: x.hackathon_id, admin_user_hackathon_rels)
+        return list(set(hackathon_ids))
+
+    def check_role(self, role):
+
+        # 0:super admin
+        # 1:comman admin
+
+        hackathon_ids = self.get_hackid_from_adminid(g.admin.id)
+        # None  or not None { has -1 or has not }
+
+        if hackathon_ids is None:
+            return False
+        else:
+            #only super admin can access
+            if role == ROLE.SUPER_ADMIN:
+                return (-1L) in (hackathon_ids)
```

Comment:
when debug I found the value showed '-1L' so I add this L
and the hackathon_ids is already a List so the () is unnecessary

because of a little afraid so do these to make the method more confidence
and  i wll fixed them ~